File: Data Structures & Algorithms/contains-duplicate-ii/submission-0.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        d = {}
        for i , num in enumerate(nums):
            if num in d:
                if abs(d[num] - i) <= k:
                    return True
            d[num] = i
        return False

        

File: Data Structures & Algorithms/subarray-sum-equals-k/submission-0.java
class Solution {
    public int subarraySum(int[] nums, int k) {
        Map<Integer, Integer> map = new HashMap<>();
        map.put(0, 1);
        int currSum = 0;
        int res =0;
        for (int num : nums){
            currSum += num;
            int diff= currSum -k;
            res += map.getOrDefault(diff,0);
            map.put(currSum, map.getOrDefault(currSum, 0) + 1);
        }
        return res;
        
    }
}